### PR TITLE
Fully functional authentication feature, allows users to comment and delete their comments if logged in

### PR DESCRIPTION
**Pull request by smruthibalajee**
**Description:** Implemented user authentication. Added a login button to nav bar that redirects to the correct link, so users can easily log in or out of the website. Created a new User class and LoginServlet to fetch the login status data. Users can only comment when logged in and they can only delete public comments made from the account through which they logged in.

**Key Features:** 

- Login button on nav bar is interactive: When the user is logged in, it displays "Logout". When the user is logged out, the button displays "Login".
login-status

- Comment limiting with submit button: All users can see the comment form, however if a user that is not logged in tries to submit a comment, they are redirected to the login page. 
comment-limit

- Comment deletion by user: Every time the page is refreshed or a new comment is submitted, the email of the current user is maintained and compared with the email associated with the comment. If they match, the user can delete the comment.
delete-comment-limit

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -104,6 +104,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     if (!userService.isUserLoggedIn()) {
       String urlToRedirectToAfterUserLogsIn = "/index.html#home";
       String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      //redirects to the login page if they try to post a comment without logging in.
```
Comment (peter14f) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: Insert one space before "redirects"

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -207,16 +208,21 @@ function fetchLoginInfo() {
 function fetchAndDisplayNumComments(num) {
     document.getElementById('comments-container').innerHTML = "";
     const dataListElement = document.getElementById('comments-container');
+    var userEmail = "";
+    
     fetch('/data?num-comments='+num).then(response => response.json()).then((data) => {
         data.forEach((comment) => {
-            dataListElement.appendChild(createCommentElement(comment));
-            displayMapComment(comment);
+            //Added to ensure that the user login email is fetched, even with page refreshes and new comments added
```
Comment (peter14f) [inline portfolio/src/main/webapp/script.js]: nit: ditto

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -232,24 +238,27 @@ function createCommentElement(comment) {
...
     commentElement.appendChild(nameElement);
     commentElement.appendChild(typeElement);
     commentElement.appendChild(locationElement);
-    commentElement.appendChild(msgElement);
-    commentElement.appendChild(brElement);
-    commentElement.appendChild(deleteButtonElement);
     commentElement.appendChild(brElement);
+    commentElement.appendChild(msgElement);
+
+    //Adds check to see if the email associated with the comment is the same as logged in user; only logged-in users can delete
```
Comment (peter14f) [inline portfolio/src/main/webapp/script.js]: nit: ditto

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -94,9 +94,8 @@ function initChart() {
 /** Function that is called when webpage is loaded. Created function because body onload doesn't support calling 3 functions in html */
 function onloadInit() {
     writeName(); 
-    fetchAndDisplayNumComments('5');
     initMap();
-    fetchLoginInfo();
+    fetchLoginInfoAndComments('5');
 }
 
 //variables used for initialization
```
Comment (peter14f) [inline portfolio/src/main/webapp/script.js]: nit: ditto

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -188,8 +187,8 @@ function writeGreeting() {
 }
 
 /**Function that fetches the current user's login information from the server
```
Comment (peter14f) [inline portfolio/src/main/webapp/script.js]: nit: ditto

**Review by saraholiviawu — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (smruthibalajee) [reply portfolio/src/main/webapp/script.js]: fixed!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (smruthibalajee) [reply portfolio/src/main/webapp/script.js]: fixed!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (smruthibalajee) [reply portfolio/src/main/webapp/script.js]: fixed!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (smruthibalajee) [reply portfolio/src/main/webapp/script.js]: fixed!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (smruthibalajee) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: fixed!

**Review by peter14f — APPROVED** [review]

**Pull request merged**